File: Data Structures & Algorithms/merge-two-sorted-linked-lists/submission-0.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */

func mergeTwoLists(list1 *ListNode, list2 *ListNode) *ListNode {
    var h, l *ListNode
    l1 := list1
    l2 := list2 
    for {
        if l1 == nil {
            if l2 == nil {
                break;
            } else {
                if l == nil {
                    l = l2
                    h = l
                } else {
                    l.Next = l2
                }
                l = l2
                l2 = l2.Next
            }
        } else if l2 == nil {
            if l == nil {
                l = l1
                h = l
            } else {
                l.Next = l1
            }
            l = l1
            l1 = l1.Next
        } else if l1.Val < l2.Val {
            if l == nil {
                l = l1
                h = l
            } else {
                l.Next = l1
            }
            l = l1
            l1 = l1.Next
        } else {
            if l == nil {
                l = l2
                h = l
            } else {
                l.Next = l2
            }
            l = l2
            l2 = l2.Next
        }
    }

    return h
}

File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-0.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        if (head == null) {
            return null;
        }

        int l = 0;
        ListNode p = head;
        for (;;) {
            if (p == null) {
                break;
            }

            l++;

            p = p.next;
        }

        if (n == l) {
            return head.next;
        }

        p = head;
        for (int i = 0; i < l - n - 1; i++) {
            p = p.next;
        }

        if (p.next != null) {
            p.next = p.next.next;
        }
        
        return head;
    }
}
